streaming/stream/weight_conf.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2022-2024 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Configures how a Stream is weighted."""
+
+from typing import Any, Dict, Optional, Sequence, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+from typing_extensions import Self
+
+from streaming.util.shorthand import normalize_count
+
+__all__ = ['StreamWeightConf']
+
+
+class StreamWeightConf:
+    """Configures how a Stream is weighted.
+
+    Args:
+        proportion (float, optional): The proportion of this StreamingDataset's samples that are
+            sampled from this Stream. As this is a relative measure, use ``epoch_size`` to
+            determine the absolute resulting size in samples. Defaults to ``None``.
+        repeat (float, optional): Stream size multiplier, aka number of times to see each of this
+            Stream's samples per epoch. Defaults to ``None``.
+        choose (str | int, optional): Stream size, aka number of samples to draw from this Stream
+            per epoch. Defaults to ``None``.
+        kwargs (Dict[str, Any]): Any unsupported (for forward compat) or deprecated args.
+    """
+
+    def __init__(
+        self,
+        *,
+        proportion: Optional[float] = None,
+        repeat: Optional[float] = None,
+        choose: Optional[Union[str, int]] = None,
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        has_proportion = proportion is not None
+        has_repeat = repeat is not None
+        has_choose = choose is not None
+        if not (0 <= has_proportion + has_repeat + has_choose <= 1):
```

Comment:
Should reflect this condition in the API arguments.

Reply:
Fixed